recipes/libpng/all/conanfile.py

```diff
@@ -51,31 +66,23 @@ def _patch(self):
                                       'COMMAND "${CMAKE_COMMAND}" -E copy_if_different $<TARGET_LINKER_FILE_NAME:${S_TARGET}> $<TARGET_LINKER_FILE_DIR:${S_TARGET}>/${DEST_FILE}',
                                       'COMMAND "${CMAKE_COMMAND}" -E copy_if_different $<TARGET_LINKER_FILE_DIR:${S_TARGET}>/$<TARGET_LINKER_FILE_NAME:${S_TARGET}> $<TARGET_LINKER_FILE_DIR:${S_TARGET}>/${DEST_FILE}')
 
-        if self.settings.build_type == 'Debug':
-            tools.replace_in_file(os.path.join(self._source_subfolder, 'libpng.pc.in'),
-                                  '-lpng@PNGLIB_MAJOR@@PNGLIB_MINOR@',
-                                  '-lpng@PNGLIB_MAJOR@@PNGLIB_MINOR@d')
-
     def _configure_cmake(self):
-        cmake = CMake(self)
-        cmake.definitions["PNG_TESTS"] = "OFF"
-        cmake.definitions["PNG_SHARED"] = self.options.shared
-        cmake.definitions["PNG_STATIC"] = not self.options.shared
-        cmake.definitions["PNG_DEBUG"] = "OFF" if self.settings.build_type == "Release" else "ON"
-        if self.settings.os == "Emscripten":
-            cmake.definitions["PNG_BUILD_ZLIB"] = "ON"
-            cmake.definitions["M_LIBRARY"] = ""
-            cmake.definitions["ZLIB_LIBRARY"] = self.deps_cpp_info["zlib"].libs[0]
-            cmake.definitions["ZLIB_INCLUDE_DIR"] = self.deps_cpp_info["zlib"].include_paths[0]
-        if self.settings.os == "Macos":
-            if 'arm' in self.settings.arch:
-                cmake.definitions["PNG_ARM_NEON"] = "on"
-        if self.options.api_prefix:
-            cmake.definitions["PNG_PREFIX"] = self.options.api_prefix
-        if self.settings.os == "Macos" and self.settings.arch == "armv8":
-            cmake.definitions["CMAKE_SYSTEM_PROCESSOR"] = "aarch64"
-        cmake.configure()
-        return cmake
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["PNG_TESTS"] = False
+        self._cmake.definitions["PNG_SHARED"] = self.options.shared
+        self._cmake.definitions["PNG_STATIC"] = not self.options.shared
+        self._cmake.definitions["PNG_DEBUG"] = self.settings.build_type == "Debug"
+        if tools.is_apple_os(self.settings.os):
+            if "arm" in self.settings.arch:
+                # FIXME: Neon should work on iOS (but currently if leads to undefined symbols), see if autotools build is better
```

Comment:
```suggestion
                # FIXME: Neon should work on iOS (but currently it leads to undefined symbols), see if autotools build is better
```

Reply:
yes I saw that after pushing the last commit, but was too lazy to fix the typo 😩